gnc/sub8_perception/include/sub8_pcl/cv_tools.hpp

```diff
@@ -20,6 +26,28 @@ typedef std::vector<cv::Point> Contour;
 // Compute the centroid of an OpenCV contour (Not templated)
 cv::Point contour_centroid(Contour& contour);
 
+// Filter a histogram of type cv::MatND generated by cv::calcHist using a gaussian kernel
+cv::MatND smooth_histogram(const cv::MatND &histogram, size_t filter_kernel_size = 3, float sigma = 1.0);
+
+// Generate a one-dimensional gaussian kernel given a kernel size and it's standard deviation (sigma)
+std::vector<float> generate_gaussian_kernel_1D(size_t kernel_size = 3, float sigma = 1.0);
+
+// Finds positive local maxima greater than (global maximum * thresh)
+std::vector<cv::Point> find_local_maxima(const cv::MatND &histogram,  float thresh_multiplier);
+
+// Finds negative local minima less than (global minimum * thresh)
+std::vector<cv::Point> find_local_maxima(const cv::MatND &histogram,  float thresh_multiplier);
+
+// Selects the mode of a multi-modal distribution closest to a given target value
+unsigned int select_hist_mode(std::vector<cv::Point> &histogram_modes, unsigned int target);
+
+// Takes in a grayscale image and segments out a semi-homogenous foreground object with
+// pixel intensities close to <target>. Tuning of last three parameters may imrove
+// results but default values should work well in most cases.
+void statistical_image_segmentation(const cv::Mat &src, cv::Mat &dest, 
+		const int hist_size, const float** ranges, const int target, std::string image_name,
```

Comment:
Minor formatting - don't bother fixing, it will be auto-fixed the next time somebody clang-formats